libnmstate/nm/route.py

```diff
@@ -0,0 +1,125 @@
+    Query running routes
+    The acs_and_ip_cfgs should be generate to generate a tuple:
+        NM.NM.ActiveConnection, NM.IPConfig
+    """
+    routes = []
+    for (active_connection, ip_cfg) in acs_and_ip_cfgs:
+        if not ip_cfg.props.routes:
+            continue
+        iface_name = _get_iface_name(active_connection)
+        if not iface_name:
+            raise NmstateInternalError(
+                'Got connection {} has not interface name'.format(
+                    active_connection.get_id()))
+        for nm_route in ip_cfg.props.routes:
+            table_id = _get_per_route_table_id(
+                nm_route,
+                iplib.KERNEL_MAIN_ROUTE_TABLE_ID)
+            route_entry = _nm_route_to_route(
+                nm_route,
+                table_id,
+                iface_name)
+            if route_entry:
+                routes.append(route_entry)
+    routes.sort(key=itemgetter(Route.TABLE_ID,
+                               Route.NEXT_HOP_INTERFACE,
+                               Route.DESTINATION))
+    return routes
+
+
+def get_config(acs_and_ip_profiles):
+    """
+    Query running routes
+    The acs_and_ip_profiles should be generate to generate a tuple:
+        NM.NM.ActiveConnection, NM.SettingIPConfig
+    """
+    routes = []
+    for (active_connection, ip_profile) in acs_and_ip_profiles:
+        if not ip_profile.props.routes:
+            continue
+        iface_name = _get_iface_name(active_connection)
+        if not iface_name:
+            raise NmstateInternalError(
+                'Got connection {} has not interface name'.format(
+                    active_connection.get_id()))
+        default_table_id = _get_cfg_default_route_table_id(ip_profile)
+        # NM supports multiple route table in single profile:
+        #   https://bugzilla.redhat.com/show_bug.cgi?id=1436531
+        # The `ipv4.route-table` and `ipv6.route-table` will be the default
+        # table id for static routes and auto routes. But each static route can
+        # still specify route table id.
+        for nm_route in ip_profile.props.routes:
+            table_id = _get_per_route_table_id(nm_route, default_table_id)
+            route_entry = _nm_route_to_route(
+                nm_route,
+                table_id,
+                iface_name)
+            if route_entry:
+                routes.append(route_entry)
+    routes.sort(key=itemgetter(Route.TABLE_ID,
+                               Route.NEXT_HOP_INTERFACE,
+                               Route.DESTINATION))
+    return routes
+
+
+def _get_cfg_default_route_table_id(ip_profile):
+    table_id = iplib.KERNEL_MAIN_ROUTE_TABLE_ID
+    if ip_profile and ip_profile.props.route_table != NM_MAIN_ROUTE_TABLE_ID:
+        table_id = ip_profile.props.route_table
+    return table_id
+
+
+def _get_per_route_table_id(nm_route, default_table_id):
+    table = nm_route.get_attribute('table')
+    return table.get_uint32() if table else default_table_id
```

Comment:
This seems to result in `table-id: !!python/long '255'` in the yaml, so adding a `int()` here makes sense.

Reply:
Added the `int()`.